backend/services/messaging/message_service.py

```diff
@@ -124,20 +124,32 @@ def send_message_to_all_contributors(project_id: int, message_dto: MessageDTO):
             MessageService._push_messages(messages)
 
     @staticmethod
-    def _push_messages(messages, mention=True):
+    def _push_messages(messages):
         if len(messages) == 0:
             return
 
         # Flush messages to get the id
         db.session.add_all([m["message"] for m in messages])
         db.session.flush()
 
-        if mention is False:
-            db.session.commit()
-            return
-
         for i, message in enumerate(messages):
             user = message.get("user")
+            obj = message.get("message")
+            if (
+                user.mentions_notifications is False
+                and obj.message_type == MessageType.MENTION_NOTIFICATION.value
+            ):
+                continue
+            if (
+                user.projects_notifications is False
+                and obj.message_type == MessageType.PROJECT_ACTIVITY_NOTIFICATION.value
```

Comment:
The intention of https://github.com/hotosm/tasking-manager/issues/3356 was to modify the behaviour only to Mention notifications. This change extends to projects and then to task comments and project chat. We could discuss that possibility, but I think it can also generate a lot of noise to mappers.

Reply:
new changes looks good